elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreIntegrationTest.java

```diff
@@ -169,6 +169,222 @@ public void havingFilterTest() throws Exception {
         runQueryWithExpectedResult(graphQLRequest, expected);
     }
 
+    /**
+     * Test the case that a where clause is promoted into having clause.
+     * @throws Exception
+     */
+    @Test
+    public void wherePromotionTest() throws Exception {
+        String graphQLRequest = document(
```

Comment:
@hellohanchen @SushrutV We had discussed about not promoting where clauses into havings because it allows things that simply don't make sense.    @hellohanchen Do you remember the context for that discussion?